network.go

```diff
@@ -852,6 +852,10 @@ func (n *network) ipamAllocate() ([]func(), error) {
 				if ip = net.ParseIP(v); ip == nil {
 					return nil, types.BadRequestErrorf("non parsable secondary ip address %s (%s) passed for network %s", k, v, n.Name())
 				}
+				if !d.Pool.Contains(ip) {
```

Comment:
You are right. I now remember we changed that.
So you are suggesting we should fail.
I am ok with that.